makefile

```diff
@@ -391,6 +395,15 @@ test:
 	   if [ 0 -ne $${ERR} ]; then echo "Some tests failed."; exit 1; \
 	   else echo "All tests passed."; fi
 
+unittest: lib
+	@echo "Unit testing the MFEM library. This may take a while..."
```

Comment:
> This may take a while...

I'd recommend separating the build phase of the unit tests from the run/test phase of the unit tests.
You can control the former through a config option, and should be able to build them in parallel.

Actually running the tests should be fairly quick.

Reply:
In the event that the combined build and test suite execution time starts approaching the 50 minute Travis CI limit for public GitHub repos, you might also be able to shave some time off the build step using `ccache`, which is especially helpful for incremental rebuilds as part of PRs.